library/ssl_tls.c

```diff
@@ -7496,8 +7486,24 @@ static int ssl_tls12_populate_transform( mbedtls_ssl_transform *transform,
                                     ciphersuite_info->cipher ) );
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
     }
+    ssl_mode = mbedtls_ssl_get_base_mode(
+                   mbedtls_cipher_info_get_mode( cipher_info ) );
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
+    if( ssl_mode == MBEDTLS_SSL_MODE_AEAD )
+    {
+        transform->taglen =
+            ciphersuite_info->flags & MBEDTLS_CIPHERSUITE_SHORT_TAG ? 8 : 16;
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        mbedtls_ssl_cipher_to_psa( ciphersuite_info->cipher, transform->taglen,
+                                   &alg, &key_type, &key_bits );
```

Comment:
Yes, as noted below, the code changes highlight where the code could be simplied.

I did not specify sufficiently that the optimization is more a code-flow simplification, rather than a performance optimization.

If I understand the rest of the code correctly, the change you suggested above applied only to AEAD.

Reply:
:rocket: